packages/project-management-automation/lib/index.js

```diff
@@ -25,8 +25,7 @@ const automations = [
 		task: ifNotFork( addFirstTimeContributorLabel ),
 	},
 	{
-		event: 'pull_request',
-		action: 'closed',
+		event: 'push',
 		task: ifNotFork( addMilestone ),
```

Comment:
Is `ifNotFork` necessary anymore?

Reply:
Yes, why wouldn't it be?

Reply:
It doesn't seem we would ever have a scenario where we would receive a `push` event and also not immediately bail at the [first condition checking the `master` branch](https://github.com/WordPress/gutenberg/pull/20058/files#diff-c856bdd8bfe0b3c6917df8184fbab9adR39) (i.e. I can't see how an error would ever occur if the `ifNotFork` was not here). I suppose it might still be better to stop it as early as possible to avoid tying it to this specific implementation.

Reply:
Forks can also have branches named `master`.

Reply:
Are we receiving `push` events for those?

Reply:
Yes, someone could make a PR from `fork/master`.

Reply:
Hm, ok. I think I may have been expecting that [this condition](https://github.com/WordPress/gutenberg/pull/20058/files#diff-c856bdd8bfe0b3c6917df8184fbab9adR39-R42) would somehow already be verifying that the commit happened on `wordpress/gutenberg`'s `master`, but I see now that's not the case.

But! `ifNotFork` was implemented for pull request events, and I think would need to be updated to support this.

https://github.com/WordPress/gutenberg/blob/cf1a6d06b0f047c5942af6190d9789b4509323f8/packages/project-management-automation/lib/if-not-fork.js#L18-L19

Notably, `payload.pull_request` doesn't exist for `push` events.

https://developer.github.com/v3/activity/events/types/#pushevent

Reply:
Then it's better to just not use it. I've removed it. No one actually makes PRs from `master` anyway. We can live without milestones for those edge cases.

Reply:
> Then it's better to just not use it. I've removed it. No one actually makes PRs from `master` anyway. We can live without milestones for those edge cases.

I don't know how to search for specific examples to share, but I've definitely seen pulls from forked `master` before. That said, the combination of this _and_ a commit message which matches the `(#PR)` pattern to get to the point of adding the milestone would be exceedingly rare, yes. And the result would merely be a failing action, nothing too worrisome.

We might want to revisit this in the future to make the logic more durable and to support `ifNotFork` usage for non-`pull_request` events (especially given the fact the name is generic enough to easily misinterpret that it ought be supported), but I don't think it's a blocker.